neo4j-store/src/main/kotlin/org/ostelco/prime/storage/graph/Neo4jStore.kt

```diff
@@ -1163,18 +1173,29 @@ object Neo4jStoreSingleton : GraphStore {
                         .bind()
 
                 if (product.price.amount > 0) {
-                    val (chargeId, invoiceId) = when (product.paymentType) {
-                        SUBSCRIPTION -> {
-                            val subscriptionDetailsInfo = purchasePlan(
+                    val purchaseRecord = when (product.paymentType) {
```

Comment:
This is a very long if-branch.  Consider making a method with a descriptive name (a scope-local one if convenient). This will increase the readability of the code.